### ISSUE TITLE
Allow "--help" without command

### ISSUE THREAD

**Issue by Jan200101** [ziglang/zig#7083]
Zig doesn't maintain its own manpage so help2man is the best option for package maintainers out there but Zig does not allow --help without a command specified as an argument without returning an error which is garbage output for help2man

**Comment by nektro** [comment]
Zig does indeed have `--help` on the base and various commands.

```
❯ zig --help
info: Usage: zig [command] [options]

Commands:

  build            Build project from build.zig
  build-exe        Create executable from source or object files
  build-lib        Create library from source or object files
  build-obj        Create object from source or assembly
  cc               Use Zig as a drop-in C compiler
  c++              Use Zig as a drop-in C++ compiler
  env              Print lib path, std path, compiler id and version
  fmt              Parse file and render in canonical zig format
  init-exe         Initialize a `zig build` application in the cwd
  init-lib         Initialize a `zig build` library in the cwd
  libc             Display native libc paths file or validate one
  run              Create executable and run immediately
  translate-c      Convert C code to Zig code
  targets          List available compilation targets
  test             Create and run a test build
  version          Print version number and exit
  zen              Print zen of zig and exit

General Options:

  --help           Print command-specific usage
```

**Comment by Jan200101** [comment]
there is an added error that you might have missed
```
sentry:zig-linux-x86_64-0.7.0+f0b1b74d2/ $ ./zig version
0.7.0+f0b1b74d2

sentry:zig-linux-x86_64-0.7.0+f0b1b74d2/ $ ./zig --help
info: Usage: zig [command] [options]

Commands:

  build            Build project from build.zig
  build-exe        Create executable from source or object files
  build-lib        Create library from source or object files
  build-obj        Create object from source or assembly
  cc               Use Zig as a drop-in C compiler
  c++              Use Zig as a drop-in C++ compiler
  env              Print lib path, std path, compiler id and version
  fmt              Parse file and render in canonical zig format
  init-exe         Initialize a `zig build` application in the cwd
  init-lib         Initialize a `zig build` library in the cwd
  libc             Display native libc paths file or validate one
  run              Create executable and run immediately
  translate-c      Convert C code to Zig code
  targets          List available compilation targets
  test             Create and run a test build
  version          Print version number and exit
  zen              Print zen of zig and exit

General Options:

  --help           Print command-specific usage

error: unknown command: --help
```
if no subcommand or flag is specified you'd get a 
```
error: expected command argument
```

**Comment by Mouvedia** [comment]
related: #715

**Comment by Jan200101** [comment]
After checking the source code it seems exactly the thing I am asking for already exists but is undocumented (or maybe the dashes were forgotten)

https://github.com/ziglang/zig/blob/master/src/main.zig#L198-L200
```zig
    } else if (mem.eql(u8, cmd, "help")) {
        try io.getStdOut().writeAll(usage);
    } else {
```

**Comment by Jan200101** [comment]
Fixed in da84ef2a9cd55bd3fdf8d2005a69ed9be86db689
